librsvg 빌드할 때 다음 에러가 발생을 합니다.

```
   Microsoft(R) Program Maintenance Utility 버전 14.29.30153.0
Copyright (c) Microsoft Corporation. All rights reserved.

        link /DLL /machine:x86 /DEBUG /LTCG /opt:ref /libpath:C:\gtk-build\gtk\Win32\release\lib vs16\release\Win32\obj\rsvg_c_api\i686-pc-windows-msvc\release\librsvg.lib pangocairo-1.0.lib                       pango-1.0.lib                   cairo-gobject.lib               gdk_pixbuf-2.0.lib      gio-2.0.lib             gobject-2.0.lib    glib-2.0.lib             cairo.lib               harfbuzz.lib    freetype.lib    libxml2.lib     intl.lib                libxml2.lib                     advapi32.lib       userenv.lib                      ws2_32.lib      /implib:vs16\release\Win32\rsvg-2.0.lib         -out:vs16\release\Win32\rsvg-2.0-vs16.dll /def:vs16\release\Win32\librsvg\librsvg.def @C:\Users\ccr\AppData\Local\Temp\nmB454.tmp
Microsoft (R) Incremental Linker Version 14.29.30153.0
Copyright (C) Microsoft Corporation.  All rights reserved.

vs16\release\Win32\librsvg\_rsvg_dummy.obj
   vs16\release\Win32\rsvg-2.0.lib 라이브러리 및 vs16\release\Win32\rsvg-2.0.exp 개체를 생성하고 있습니다.
librsvg.lib(std-229f15d06c4fb5c2.std.45b94fb9fd9c553c-cgu.0.rcgu.o) : error LNK2001: 확인할 수 없는 외부 기호 __imp__NtCreateFile@44
librsvg.lib(std-229f15d06c4fb5c2.std.45b94fb9fd9c553c-cgu.0.rcgu.o) : error LNK2001: 확인할 수 없는 외부 기호 __imp__RtlNtStatusToDosError@4
librsvg.lib(std-229f15d06c4fb5c2.std.45b94fb9fd9c553c-cgu.0.rcgu.o) : error LNK2001: 확인할 수 없는 외부 기호 __imp__NtReadFile@36
librsvg.lib(std-229f15d06c4fb5c2.std.45b94fb9fd9c553c-cgu.0.rcgu.o) : error LNK2001: 확인할 수 없는 외부 기호 __imp__NtWriteFile@36
librsvg.lib(std-229f15d06c4fb5c2.std.45b94fb9fd9c553c-cgu.0.rcgu.o) : error LNK2001: 확인할 수 없는 외부 기호 __imp__BCryptGenRandom@16
vs16\release\Win32\rsvg-2.0-vs16.dll : fatal error LNK1120: 5개의 확인할 수 없는 외부 참조입니다.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\bin\HostX86\x86\link.EXE"' : '0x460' 반환 코드입니다.
Stop.
Traceback (most recent call last):
  File "C:\gtk-build\github\gvsbuild\gvsbuild\utils\builder.py", line 480, in build
    if self.__build_one(p):
  File "C:\gtk-build\github\gvsbuild\gvsbuild\utils\builder.py", line 606, in __build_one
    skip_deps = proj.build()
  File "C:\gtk-build\github\gvsbuild\gvsbuild\projects.py", line 1221, in build
    self.exec_vs(r'nmake install -f Makefile.vc CFG=%(configuration)s PYTHON="%(python_dir)s\python.exe" PREFIX="%(gtk_dir)s"')
  File "C:\gtk-build\github\gvsbuild\gvsbuild\utils\base_project.py", line 109, in exec_vs
    self.builder.exec_vs(cmd, working_dir=self._get_working_dir(), add_path=add_path)
  File "C:\gtk-build\github\gvsbuild\gvsbuild\utils\builder.py", line 854, in exec_vs
    self.__execute(self.__sub_vars(cmd), working_dir=working_dir, add_path=add_path, env=self.vs_env)
  File "C:\gtk-build\github\gvsbuild\gvsbuild\utils\builder.py", line 931, in __execute
    subprocess.check_call(args, cwd=working_dir, env=env, shell=True)
  File "C:\Python36\lib\subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command 'nmake install -f Makefile.vc CFG=release PYTHON="C:\Python37\python.exe" PREFIX="C:\gtk-build\gtk\Win32\release"' returned non-zero exit status 2.
Error: librsvg build failed
```

--- a/README.md
+++ b/README.md
@@ -21,10 +21,11 @@ HexChat developers decided that their script should focus on their specific need
 
     * [Visual Studio for Windows Desktop](http://www.visualstudio.com/downloads) - 2013, 2015 and 2017 are currently supported.
     * [msys2](https://msys2.github.io/)
-    * [Python 3.6](https://www.python.org/ftp/python/3.6.2/python-3.6.2-amd64.exe)
+    * [Python 3.7](https://www.python.org/ftp/python/3.7.0/python-3.7.0-amd64.exe)
+    * [CMake 3.28](https://github.com/Kitware/CMake/releases/download/v3.28.0-rc5/cmake-3.28.0-rc5-windows-x86_64.msi)
 
 1. Environment PATH System Variables
-    * C:\Users\ccr\AppData\Local\Programs\Python\Python36
+    * C:\Python37
 
 1. Follow the instructions on the msys2 page to update the core packages. The needed packages for the script (make, diffutils, ...) are download and installed automatically if not presents in the msys2 installation.
 
@@ -68,9 +69,9 @@ HexChat developers decided that their script should focus on their specific need
     ```
 1. To build CMS
 
-    ```
-    python build.py build lz4 openssl gtk librsvg libssh libuv libcurl libjpeg-turbo json-c -p x86 -c release --vs-ver 15
-    python build.py build lz4 openssl gtk librsvg libssh libuv libcurl libjpeg-turbo json-c -p x64 -c release --vs-ver 15
+    ``
+    python build.py build lz4 openssl gtk librsvg libssh libuv libcurl libjpeg-turbo json-c -p x86 -c release --vs-ver 16
+    python build.py build lz4 openssl gtk librsvg libssh libuv libcurl libjpeg-turbo json-c -p x64 -c release --vs-ver 16
     ```
 
 1. When the script is done, your GTK+ stack will be found under _C:\gtk-build\gtk_. Enjoy!

--- a/gvsbuild/projects.py
+++ b/gvsbuild/projects.py
@@ -1210,7 +1210,7 @@ class Project_librsvg(Tarball, Project, _MakeGir):
             archive_url = 'http://ftp.acc.umu.se/pub/GNOME/sources/librsvg/2.52/librsvg-2.52.0.tar.xz',
             hash = 'bd821fb3e16494b61f5185addd23b726b064f203122b3ab4b3d5d7a44e6bf393',
             dependencies = ['libcroco', 'cairo', 'pango', 'gdk-pixbuf'],
-            patches = [],
+            patches = ['001-library-link.patch'],
             )
 
     def build(self):

--- a/gvsbuild/utils/base_project.py
+++ b/gvsbuild/utils/base_project.py
@@ -124,7 +124,9 @@ class Project(object):
         """
 
         ver = self.builder.opts.vs_ver
-        if ver == '16':
+        if ver == '17':
+            dst_platform = '143'
+        elif ver == '16':
             dst_platform = '142'
         elif ver == '15':
             dst_platform = '141'

--- a/gvsbuild/utils/parser.py
+++ b/gvsbuild/utils/parser.py
@@ -99,6 +99,9 @@ def get_options(args):
         elif opts.vs_ver == "16":
             opts.vs_install_path = r'C:\Program Files (x86)\Microsoft Visual Studio\2019'
             opts._vs_path_auto = True
+        elif opts.vs_ver == "17":
+            opts.vs_install_path = r'C:\Program Files\Microsoft Visual Studio\2022'
+            opts._vs_path_auto = True
         else:
             opts.vs_install_path = r'C:\Program Files (x86)\Microsoft Visual Studio %s.0' % (opts.vs_ver,)
 
@@ -237,8 +240,8 @@ Examples:
                          help="The directory where you checked out https://github.com/wingtk/gvsbuild.git. Default is $(build-dir)\\github\\gvsbuild.")
     p_build.add_argument('--tools-root-dir',
                          help="The directory where to install the downloaded tools. Default is $(build-dir)\\tools.")
-    p_build.add_argument('--vs-ver', default='16',
-                         help="Visual Studio version 12 (vs2013), 14 (vs2015), 15 (vs2017), 16 (vs2019) etc. Default is 12.")
+    p_build.add_argument('--vs-ver', default='17',
+                         help="Visual Studio version 12 (vs2013), 14 (vs2015), 15 (vs2017), 16 (vs2019), 17 (vs2022). Default is 17.")
     p_build.add_argument('--vs-install-path',
                          help=r"The directory where you installed Visual Studio. Default is 'C:\Program Files (x86)\Microsoft Visual Studio $(vs-ver).0' (for vs-ver <= 14) " +
                          "or 'C:\Program Files (x86)\Microsoft Visual Studio\\20xx' (2017 for vs-ver 15, 2019 for vs-ver 16, ...). "

--- a/gvsbuild/utils/parser.py
+++ b/gvsbuild/utils/parser.py
@@ -237,7 +237,7 @@ Examples:
                          help="The directory where you checked out https://github.com/wingtk/gvsbuild.git. Default is $(build-dir)\\github\\gvsbuild.")
     p_build.add_argument('--tools-root-dir',
                          help="The directory where to install the downloaded tools. Default is $(build-dir)\\tools.")
-    p_build.add_argument('--vs-ver', default='15',
+    p_build.add_argument('--vs-ver', default='16',
                          help="Visual Studio version 12 (vs2013), 14 (vs2015), 15 (vs2017), 16 (vs2019) etc. Default is 12.")
     p_build.add_argument('--vs-install-path',
                          help=r"The directory where you installed Visual Studio. Default is 'C:\Program Files (x86)\Microsoft Visual Studio $(vs-ver).0' (for vs-ver <= 14) " +

--- a/patches/librsvg/001-library-link.patch
+++ b/patches/librsvg/001-library-link.patch
@@ -1,0 +1,24 @@
+diff -bru a/win32/config-msvc.mak b/win32/config-msvc.mak
+--- a/win32/config-msvc.mak	2021-09-16 10:15:11.000000000 +0900
++++ b/win32/config-msvc.mak	2023-11-28 13:33:01.222761400 +0900
+@@ -118,6 +118,8 @@
+ 	libxml2.lib			\
+ 	advapi32.lib			\
+ 	userenv.lib			\
++	ntdll.lib			\
++	bcrypt.lib			\
+ 	ws2_32.lib
+ 
+ RSVG_PIXBUF_LOADER_CFLAGS =			\
+diff -bru a/win32/config-msvc.mak.in b/win32/config-msvc.mak.in
+--- a/win32/config-msvc.mak.in	2021-09-14 09:16:20.000000000 +0900
++++ b/win32/config-msvc.mak.in	2023-11-28 13:32:55.955415300 +0900
+@@ -118,6 +118,8 @@
+ 	libxml2.lib			\
+ 	advapi32.lib			\
+ 	userenv.lib			\
++	ntdll.lib			\
++	bcrypt.lib			\
+ 	ws2_32.lib
+ 
+ RSVG_PIXBUF_LOADER_CFLAGS =			\

